src/goto-programs/remove_function_pointers.h

```diff
@@ -16,12 +16,23 @@ Date: June 2003
 
 #include <util/irep.h>
 
+#include <map>
+
+#include "remove_const_function_pointers.h"
+
 class goto_functionst;
 class goto_programt;
 class goto_modelt;
 class message_handlert;
 class symbol_tablet;
 
+using possible_fp_targetst = remove_const_function_pointerst::functionst;
+using possible_fp_targets_mapt = std::map<irep_idt, possible_fp_targetst>;
+
+possible_fp_targets_mapt get_function_pointer_targets(
+  message_handlert &message_handler,
+  goto_modelt &goto_model);
+
 // remove indirect function calls
 // and replace by case-split
 void remove_function_pointers(
```

Comment:
One more thing; can there be a version of `remove_function_pointers` that takes in a `possible_fp_targets_mapt` as that way we have a way of producing more compact instantiations of function pointers.

Reply:
Well, that was fun: https://github.com/diffblue/cbmc/pull/4536/commits/74584f8bc0967474ac0a4bf1b0274c4250e51219.

Reply:
Sorry; that was rather more work than I had thought it would be.  Thanks for doing.